From 1817cb215d6b1f03d4f967a742bcbdde843894e5 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 4 Mar 2011 11:33:23 +0100
Subject: [PATCH] selinux: on failure "die" rather than sleep

This ensures, that plymouth is stopped and the message can be seen.
---
 modules.d/99base/selinux-loadpolicy.sh | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/modules.d/99base/selinux-loadpolicy.sh b/modules.d/99base/selinux-loadpolicy.sh
index 9959993..2a19fe1 100755
--- a/modules.d/99base/selinux-loadpolicy.sh
+++ b/modules.d/99base/selinux-loadpolicy.sh
@@ -47,17 +47,12 @@ rd_load_policy()
 
 	warn "Initial SELinux policy load failed."
 	if [ $ret -eq 3 -o $permissive -eq 0 ]; then
-	    warn "Machine in enforcing mode."
-	    warn "Not continuing"
-	    sleep 100d
+	    die "Initial SELinux policy load failed. Machine in enforcing mode. To disable selinux, add selinux=0 to the kernel command line."
 	    exit 1
 	fi
 	return 0
     elif [ $permissive -eq 0 -a "$SELINUX" != "disabled" ]; then
-	warn "Machine in enforcing mode and cannot execute load_policy."
-	warn "To disable selinux, add selinux=0 to the kernel command line."
-	warn "Not continuing"
-	sleep 100d
+	die "Machine in enforcing mode and cannot execute load_policy. To disable selinux, add selinux=0 to the kernel command line."
 	exit 1
     fi
 }
